### DISCUSSION TITLE
select_only will return an error

### DISCUSSION THREAD

**Discussion by 891806797** [SeaQL/sea-orm#823] [General]
let user_res = user_account::Entity::find()
            .select_only()
            .column(user_account::Column::Id)
            .column(user_account::Column::Username)
            .column(user_account::Column::Passwd)
            .filter(user_account::Column::Username.eq(self.username.as_str()))
            .one(&sdk.as_ref().db)
            .await;

Error message：
Query Error: no column found for name: nickname

**Comment by tyt2y3** [comment]
We need the entity file to reproduce the problem.

**Reply by 891806797** [reply]
//! SeaORM Entity. Generated by sea-orm-codegen 0.8.0

use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "user_account")]
pub struct Model {
    #[sea_orm(primary_key, unique)]
    pub id: i32,
    pub username: String,
    pub passwd: String,
    pub nickname: Option<String>,
    pub avatar: Option<String>,
    pub gender: Option<i8>,
    pub mobile: Option<String>,
    pub vx_no: Option<String>,
    pub vip_level: Option<i8>,
    pub vip_expire: Option<DateTimeUtc>,
    pub views: Option<i32>,
    pub collects: Option<i32>,
    pub status: Option<i8>,
    pub create_time: Option<DateTimeUtc>,
}

#[derive(Copy, Clone, Debug, EnumIter)]
pub enum Relation {}

impl RelationTrait for Relation {
    fn def(&self) -> RelationDef {
        panic!("No RelationDef")
    }
}

impl ActiveModelBehavior for ActiveModel {}

**Reply by 891806797** [reply]
this model generated from mysql with sea-orm-cli

**Reply by 891806797** [reply]
CREATE TABLE IF NOT EXISTS user_account
(
    id          INT NOT NULL AUTO_INCREMENT UNIQUE,
    username    VARCHAR(20) NOT NULL,
    passwd      VARCHAR(68) NOT NULL,
    nickname    VARCHAR(20)  DEFAULT '',
    avatar      VARCHAR(255) DEFAULT '',
    gender      TINYINT      DEFAULT 0,
    mobile      VARCHAR(68)  DEFAULT '',
    vx_no       VARCHAR(100) DEFAULT '',
    vip_level   TINYINT      DEFAULT 1,
    vip_expire  TIMESTAMP    DEFAULT CURRENT_TIMESTAMP,
    views       INT          DEFAULT 0,
    collects    INT          DEFAULT 0,
    status      TINYINT      DEFAULT 1,
    create_time TIMESTAMP    DEFAULT CURRENT_TIMESTAMP,
    PRIMARY KEY (id),
    KEY user_account_mobile (mobile) USING BTREE,
    KEY user_account_username (username) USING BTREE,
    KEY user_account_create_time (create_time) USING BTREE
) ENGINE = InnoDB
  AUTO_INCREMENT = 147054
  DEFAULT CHARSET = utf8mb4 COMMENT ='user_table';

**Comment by rgreinho** [comment]
I had the same problem and it was because the results could not be deserialized into the model due to some missing values that were required.

I solved the problem by creating a struct just to handle the exact query result, as described in https://www.sea-ql.org/SeaORM/docs/advanced-query/custom-select#handling-custom-selects.

So in your case it should be something like that:

```rust
#[derive(FromQueryResult, Debug, Serialize, Deserialize)]
struct LimitedUserAccount {
   id: i32,
   username: String,
   passwd: String,
}
```

And you would add:

```rust
.into_model::<LimitedUserAccount>()
```

before the `.one(&sdk.as_ref().db)` statement.

**Reply by 891806797** [reply]
thank you very much!!!
